zerver/lib/events.py

```diff
@@ -359,10 +358,7 @@ def fetch_initial_state_data(
             # The webapp doesn't use the data from here; instead,
             # it uses data from state["subscriptions"] and other
             # places.
-            if user_profile is not None:
-                state['streams'] = do_get_streams(user_profile)
-            else:
-                state['streams'] = get_web_public_streams(realm)
+            state['streams'] = [] if user_profile is None else do_get_streams(user_profile)
```

Comment:
I think the pre-patch version of this logic was more conceptually correct -- we want the `include_streams` behavior to be as expected in case we choose to add app mobile support for this sort of logged-out view, or decide to use the `streams` data structures in the webapp in the future.

Did you remove this because there's no test coverage on the `include_streams` code path, or something else?

Reply:
I noticed it because there was no coverage. Then saw that we were not using it and it adds a query to the DB.

Reply:
Added it back. I guess it didn't matter since were passing `include_streams=False`.